src/mon/OSDMonitor.cc

```diff
@@ -3346,7 +3346,8 @@ void OSDMonitor::tick()
 	}
 
 	if (g_conf->mon_osd_down_out_interval > 0 &&
-	    down.sec() >= grace) {
+	    down.sec() >= grace &&
+            !osdmap.is_destroyed(o)) {
```

Comment:
The "`osd destory`" command is introduced mainly for osd replacing process. We want to minimize the data migration, but the auto-out mechanism does quite the opposite(for data safety).

Since user is aware of what he/she is doing(osd destroy can be only manually initiated), we can definitely disable this safely.